pushy/src/main/java/com/turo/pushy/apns/proxy/ProxyLocator.java

```diff
@@ -0,0 +1,48 @@
+package com.turo.pushy.apns.proxy;
+
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.net.ProxySelector;
+import java.net.SocketAddress;
+import java.net.URI;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Helper class to look for proxy configurations.
+ *
+ * @see <a href="https://docs.oracle.com/javase/7/docs/api/java/net/doc-files/net-properties.html#Proxies">The Java documentation on proxies.</a>
+ * @since 0.14
+ */
+class ProxyLocator {
+
+    private static final Logger log = LoggerFactory.getLogger(ProxyLocator.class);
+
+    /**
+     * Returns the {@link SocketAddress} for any system-wide {@link Proxy} of a given {@link Proxy.Type} for a given {@link URI}.
+     *
+     * @param uri the URI to find any system proxy for.
+     * @param proxyType the types of proxy to find for the URI.
+     * @return the {@link SocketAddress} of any <code>Proxy</code> found, or <code>null</code> if there were none.
+     * @since 0.14
+     */
+    static SocketAddress getProxyAddressForUri(final URI uri, final Proxy.Type proxyType) {
```

Comment:
It looks like this is only called from one place. Does it make sense to move this method to `HttpProxyHandlerFactory` rather than putting it in its own class?

Reply:
That is up to you. I put it there so that it can be reused for a SOCKS proxy. I can write the code blind, but can't test it. Given how simple the code using it is, there is a decent chance it will just work.

Reply:
I'm definitely open to the SOCKS plan (and think I can create a reasonable test environment), but how would we tell the difference between SOCKS4 and SOCKS5?